src/Compilers/CSharp/Portable/BoundTree/DecisionTreeBuilder.cs

```diff
@@ -0,0 +1,570 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    /// <summary>
+    /// Helper class for (1) finding reachable and unreachable switch cases in binding, and (2)
+    /// building a decision tree for lowering.
+    /// </summary>
+    internal abstract class DecisionTreeBuilder
```

Comment:
>DecisionTreeBuilder [](start = 28, length = 19)

This class can benefit from a comment explaining the principles of building the tree.